test/test_gc.c

```diff
@@ -187,8 +187,8 @@ static char* test_gc_allocation_map_cleanup()
 static char* test_gc_mark_stack()
 {
     GarbageCollector gc_;
-    int bos;
-    gc_start_ext(&gc_, &bos, 32, 32, 0.0, DBL_MAX, DBL_MAX);
+    void *bos = __builtin_frame_address(0);
+    gc_start_ext(&gc_, bos, 32, 32, 0.0, DBL_MAX, DBL_MAX);
```

Comment:
This seems like the correct fix - can you introduce that in all `test_*` functions that initialize a `GarbageCollector` and test if that solves the issue for all test cases?

Reply:
Sure, I'll extend it to all